core/pta/device.c

```diff
@@ -23,8 +24,11 @@
 static void add_ta(uint32_t flags, const TEE_UUID *uuid, uint8_t *buf,
 		   uint32_t blen, uint32_t *pos, uint32_t rflags)
 {
-	if ((flags & TA_FLAG_DEVICE_ENUM) &&
-	    (flags & TA_FLAG_DEVICE_ENUM_SUPP)) {
+	flags &= (TA_FLAG_DEVICE_ENUM | TA_FLAG_DEVICE_ENUM_SUPP |
+		  TA_FLAG_DEVICE_ENUM_SEC_STORAGE);
+	if (flags && flags != TA_FLAG_DEVICE_ENUM &&
+	    flags != TA_FLAG_DEVICE_ENUM_SUPP &&
+	    flags != TA_FLAG_DEVICE_ENUM_SEC_STORAGE) {
```

Comment:
I think we should handle the "enumerable" bit and the dependencies separately. Then `TA_FLAG_DEVICE_ENUM_SEC_STORAGE` would be best renamed to `TA_FLAG_DEPENDS_ON_SEC_STORAGE`.
In addition, "secure storage" is ambiguous since it can be implemented either with RPMB or the REE FS, which might end up requiring tee-supplicant or not. But the TA does not necessarily know about it (it knows in case it uses `_PRIVATE_RPMB` or `_PRIVATE_REE_FS` but not if it use `_PRIVATE`, in this case only the TEE core knows).

Reply:
`TA_FLAG_DEVICE_ENUM_SEC_STORAGE` is an abstraction that selects the required stage to be enumerated. It does not cover `TEE_STORAGE_PRIVATE_REE` or `TEE_STORAGE_PRIVATE_RPMB`, that would require yet another flag.

So the flag isn't about what the TA depends on, it's the stage where it should be enumerated.

Reply:
I'll update the commit message for "core: add flag to indicate dependency on RPMB" to align better with this.